File: engine/transpile/src/test/resources/golden/SYK005/LK_メッセージ区分.py
"""SYK005 のデータ部レコード LK-メッセージ区分 を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class LK_メッセージ区分:
    """01 LK-メッセージ区分 総バイト長 2。"""

    LENGTH = 2

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(2)
        else:
            self._data = bytearray(data)
            if len(self._data) < 2:
                self._data.extend(b"\x00" * (2 - len(self._data)))

    @property
    def data(self):
        return self._data

    # LK-メッセージ区分
    def get_LK_メッセージ区分(self):
        return decode_alphanumeric(self._data, 0, 2)
    def set_LK_メッセージ区分(self, value):
        encode_alphanumeric(self._data, 0, 2, value)

File: src/engine/transpile/src/test/resources/golden/SYK005/LK_メッセージ区分.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK005 のデータ部レコード LK-メッセージ区分 を逐語対訳した自動生成コード。 */
public final class LK_メッセージ区分 {

    /** 01 LK-メッセージ区分 総バイト長 2。 */
    public static final int LENGTH = 2;

    private final byte[] data;

    public LK_メッセージ区分() {
        this.data = new byte[2];
    }

    public LK_メッセージ区分(byte[] source) {
        this.data = Arrays.copyOf(source, 2);
    }

    public byte[] data() {
        return data;
    }

    // LK-メッセージ区分
    public String get_LK_メッセージ区分() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 2);
    }
    public void set_LK_メッセージ区分(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 2, value);
    }
}
